common.js

```diff
@@ -0,0 +1,66 @@
+/* global jQuery chrome notify vaultToken querySecrets */
+/* eslint no-unused-vars: "off" */
+/* eslint "space-before-function-paren": ["error", {
+        "anonymous": "always",
+        "named": "never",
+        "asyncArrow": "always"
+    }], */
+
+// invoked after user clicks "login to vault" button, if all fields filled in, and URL passed regexp check.
+function authToVault(vaultServer, username, password, authMount) {
+  var notify = document.getElementById('notify')
+  console.debug('Login attempt: ' + vaultServer + '/v1/auth/' + authMount + '/login/' + username)
+  jQuery.ajax({
+    type: 'POST',
+    url: vaultServer + '/v1/auth/' + authMount + '/login/' + username,
+    data: JSON.stringify({ 'password': password }),
+    contentType: 'application/json',
+    dataType: 'json',
+    success: function (data) {
+      console.debug('AUTH success')
+      notify.textContent = 'AUTH SUCCESS, token: ' + data.auth.client_token
+      vaultToken = data.auth.client_token
+      chrome.storage.local.set({ 'vaultToken': data.auth.client_token }, function () { })
+      console.debug(data)
+      querySecrets(vaultServer)
+      // TODO: Use user token to generate app token with 20h validity - then use THAT token
+    },
+    error: function (data) {
+      console.debug('AUTH fail', data)
+      notify.textContent = 'ERROR: ' + JSON.stringify(data)
+    }
+  })
+}
+
+// verify URL function (copy pasted from google, standard one)
+function checkURL(value) {
+  var urlregex = new RegExp("^(http|https)://([a-zA-Z0-9.-]+(:[a-zA-Z0-9.&amp;%$-]+)*@)*((25[0-5]|2[0-4][0-9]|[0-1]{1}[0-9]{2}|[1-9]{1}[0-9]{1}|[1-9]).(25[0-5]|2[0-4][0-9]|[0-1]{1}[0-9]{2}|[1-9]{1}[0-9]{1}|[1-9]|0).(25[0-5]|2[0-4][0-9]|[0-1]{1}[0-9]{2}|[1-9]{1}[0-9]{1}|[1-9]|0).(25[0-5]|2[0-4][0-9]|[0-1]{1}[0-9]{2}|[1-9]{1}[0-9]{1}|[0-9])|([a-zA-Z0-9-]+.)*[a-zA-Z0-9-]+.(com|edu|gov|int|mil|net|org|biz|arpa|info|name|pro|aero|coop|museum|[a-zA-Z]{2}))(:[0-9]+)*(/($|[a-zA-Z0-9.,?'\\+&amp;%$#=~_-]+))*$")
```

Comment:
This one seems buggy. Having a list of tlds seems not a good idea, there are way to many now but since the regex doesn't work as intended it does not really check the tld - so it works. Sometimes.

Examples:
 * it doesn't accept `http://abc.xyz:123'` which it should
 * it does accept `http://abcd` but not `http://abc`

Reply:
I would leave that up to the user, if he enters a wrong URL and things don't connect, it's his problem, just need to display an error, and that's it. 
Up to you though, just my 2 cents.

Reply:
actually I took that over from @Dmitry1987 - I'm happy to remove this part

Reply:
https://github.com/zeichenanonym/vault-chrome-extension/commit/a97256bac5aaeee71bda1464ac268236073921e7 resolves this